### PR TITLE
Remove all usage of $CFG->additionalhtmltopofbody

### PR DESCRIPTION
**Pull request by ScottVerbeek**
This Pull requests contains the Commits of both https://github.com/catalyst/moodle-local_envbar/pull/155 and https://github.com/catalyst/moodle-local_envbar/pull/157

### CODE REVIEW

File: tests/lib_test.php [hunk 1]
```diff
@@ -111,14 +111,14 @@ public function test_inject() {
         $data->showtext = 'Test Inject';
         envbarlib::update_envbar($data);
 
-        envbarlib::reinject();
+        envbarlib::reset_injectcalled();
         self::assertStringContainsString('<style>', $CFG->additionalhtmltopofbody);
```
Comment (brendanheywood) [inline tests/lib_test.php]: this should be touched

File: tests/lib_test.php [hunk 2]
```diff
@@ -144,7 +144,7 @@ public function test_clean_envbars() {
         $data->showtext = 'Test Inject';
         envbarlib::update_envbar($data);
 
-        $injected = envbarlib::reinject();
+        $injected = envbarlib::reset_injectcalled();
 
         // An envbar is not in the database get_config('additionalhtmltopofbody').
         $additional = get_config('core', 'additionalhtmltopofbody');
```
Comment (brendanheywood) [inline tests/lib_test.php]: and here, I'd grep for additionalhtmltopofbody, it should not exist in the entire plugin at all after this patch

Comment (ScottVerbeek) [reply tests/lib_test.php]: Okay, I'll have a do this grep now

File: lib.php [hunk 3]
```diff
@@ -43,7 +42,7 @@ function local_envbar_extend_navigation() {
     global $CFG;
 
     if (isset($CFG->branch) && $CFG->branch < 33) {
-        envbarlib::inject();
+        return envbarlib::get_inject_code();
```
Comment (brendanheywood) [inline lib.php]: this whole callback should be removed

File: lib.php [hunk 4]
```diff
@@ -54,10 +53,17 @@ function local_envbar_extends_navigation() {
     global $CFG;
 
     if ($CFG->branch < 33) {
-        envbarlib::inject();
+        return envbarlib::get_inject_code();
```
Comment (brendanheywood) [inline lib.php]: this callback should be removed

File: tests/lib_test.php (hunk 2 above)
Comment (ScottVerbeek) [reply tests/lib_test.php]: A rgrep gives no results:
```
~local/envbar$ rgrep 'additionaltopofhtml'
~local/envbar$ 
```

File: lib.php [hunk 5]
```diff
@@ -32,30 +32,14 @@
  * This hook was introduced in moodle 3.3.
  */
 function local_envbar_before_http_headers() {
-
-    envbarlib::inject();
+    return envbarlib::get_inject_code();
```
Comment (brendanheywood) [inline lib.php]: I think this one is also redundant

Comment (ScottVerbeek) [reply lib.php]: This has now been removed

File: version.php [hunk 6]
```diff
@@ -29,7 +29,7 @@
     die('Direct access to this script is forbidden.'); // It must be included from a Moodle page.
 }
 
-$plugin->version   = 2020112000;      // The current plugin version (Date: YYYYMMDDXX).
+$plugin->version   = 2021011910;      // The current plugin version (Date: YYYYMMDDXX).
 $plugin->release   = 2020112000;      // Same as version
```
Comment (brendanheywood) [inline version.php]: please keep release in lock step with version

**Comment by brendanheywood** [comment]
This is good to go, merge when green

**Pull request merged**